client/components/domains/register-domain-step/index.jsx

```diff
@@ -268,6 +270,21 @@ class RegisterDomainStep extends React.Component {
 		}
 	}
 
+	checkForBloggerPlan() {
+		const isBloggerPlan =
+			( this.props.selectedSite && isBlogger( this.props.selectedSite.plan ) ) ||
```

Comment:
Checking for `this.props.selectedSite` doesn't ensure that `this.props.selectedSite.plan` is defined (though it should be).  You might want to consider using something like `get( this.props, 'selectedSite.plan', false )` to check for this value since passing in an object without the required attributes will cause `assertValidProduct()` to fail.